src/components/sidebar/routePathView/RoutePathView.tsx

```diff
@@ -135,16 +202,30 @@ class RoutePathView extends React.Component<IRoutePathViewProps, IRoutePathViewS
             );
         }
         if (!this.props.routePathStore!.routePath) return null;
+
+        const isSaveButtonDisabled = this.state.isEditingDisabled
+        || !this.props.routePathStore!.isDirty
+        || !this.props.routePathStore!.isGeometryValid
+        || !this.isFormValid();
+
         return (
             <div className={s.routePathView}>
                 <RoutePathHeader
+                    hasModifications={this.props.routePathStore!.isDirty}
                     routePath={this.props.routePathStore!.routePath!}
-                    isAddingNew={this.props.isAddingNew}
+                    isNewRoutePath={this.props.isNewRoutePath}
```

Comment:
Sometime we use this.props.isNewRoutePath and sometimes this.routePathIsNew(), chose one way and use that consistently. Personally I think

```
   private routePathIsNew = () => {
        return this.props.isNewRoutePath;
    }
```

function could be removed and just use the prop directly. 

this.props.isNewRoutePath is not exactly much longer than this.routePathIsNew()